pkg/kmodule/kmodule.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2017-2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package kmodule
+
+import (
+	"errors"
+	"io"
+	"os"
+)
+
+// Loader is the interface for module setup, loading, and listing.
+type Loader interface {
+	Init(image []byte, opts string) error
+	FileInit(f *os.File, opts string, flags uintptr) error
+	Probe(name string, modParams string) error
+	Delete(name string, flags uintptr) error
+	List(w io.Writer) error
```

Comment:
Should we consider returning a slice of some struct here instead of writing the output to io.Writer, and having cmd/core/lsmod change it to a table instead?

I understand that today the only use of List() is to print human-readable output to callers of cmd/core/lsmod ... but I can imagine that getting a list of installed modules is something that maybe the robots/code will need in the future, so returning the data in a structured fashion here might be helpful in the future?

Reply:
Thing is, in typical Linux fashion, proc/modules is human readable and must be parsed. I struggled about the return value, but in the end, felt it should go this way. Note that it greatly simplifies the Darwin code too. Happy to change it but ... /proc/modules is just a wall of text. Plus, don't forget, what *bsd, Darwin, and Linux will return will be very, very different.

Reply:
That said, this list should die, and the type should implement io.Reader, so will do that tmw.

Reply:
see what you thiink this time around. I expect it will need another iteration.